lockfile/src/parsers/go_sum.rs

```diff
@@ -0,0 +1,57 @@
+use std::ops::Not;
+
+use nom::bytes::complete::take_until;
+use nom::character::complete::space0;
+use nom::combinator::recognize;
+use nom::multi::many1;
+use phylum_types::types::package::{PackageDescriptor, PackageType};
+
+use super::{take_till_line_end, Result};
+
+pub fn parse(input: &str) -> Result<&str, Vec<PackageDescriptor>> {
+    let (input, pkg_options) = many1(package)(input)?;
+    let pkgs = pkg_options.iter().flatten().cloned().collect::<Vec<_>>();
+
+    Ok((input, pkgs))
+}
```

Comment:
This is still a very permissive parser. It accepts any file with at least 3 words in the first line. In fact, it successfully parses every file in `tests/fixtures` (the `pip` and `gradle` parsers also accomplish this... But that's a separate issue).

A few ways that it could be made more strict:

1. Check for EOF at the end of the parser
2. Check the format of the hash (i.e., `preceded(tag("h1"), hex_digit1)`)
3. Check for extra data at the end of each line